src/filter.rs

```diff
@@ -452,6 +461,82 @@ pub(crate) fn unfilter(
                 *curr = curr.wrapping_add(above);
             }
         }
+        Avg if previous.is_empty() => match tbpp {
```

Comment:
I wonder if `roundtrip_ascending_previous_line` and `roundtrip` tests (below in the same source file) could be duplicated and/or tweaked to explicitly cover round-tripping when there is no previous row?